### ISSUE TITLE
Completion-option: specify how often to query completion providers

### ISSUE THREAD

**Issue by runeksvendsen** [microsoft/vscode#92713]
# Completion scenario

1. I type in "g"
2. A completion provider returns a bunch of completion items all starting with "g"
3. I type in an "m" (so I have typed in the word "gm")
4. Result: VS Code will take the suggestions returned by the completion provider in step 2 and filter them (rather than query the completion provider again with the query "gm")

As far as I can see (from manual testing) VS Code will keep doing step 4 above until it has filtered off all items returned by the initial query (step 2), at which point it will query the completion provider again for a fresh list of completion items.

# Feature request

I'm looking for a VS Code option that controls the extent to which VS Code will re-query the completion provider when the user types in additional letters. In other words, I want to control the extent to which VS Code prefers **a)** doing client side filtering on completion items returned by an earlier query, and **b)** re-querying the completion provider with the new input text.

# Notes

Re-querying the completion provider only really makes sense when it returns the maximum number of results it can return. I.e. if my completion provider will return, at most, 50 items (let's call this setting `maxResultCount=50`), then re-querying makes sense only when it returns this maximum number of results (since it means there might be more relevant results that were not included).

Conversely, if the completion provider returns only, say, 12 results from a query (with `maxResultCount=50`), then client-side filtering of previous results is fine, as the new query would just reduce the number of results from 12 to less.

So, in other words, the above "Feature request"-setting is only relevant when a query is not specific enough to return fewer items than the completion provider's `maxResultCount` setting. So, perhaps, it's necessary to:

1. Change the VS Code `CompletionItemProvider` API to (optionally) allow specifying the maximum number of results returned (`maxResultCount`)
2. Add a VS Code completion option that controls whether a completion provider is re-queried when the number of results returned by a previous query equals `maxResultCount`

### VS Code version info
Version: 1.43.0
Commit: 78a4c91400152c0f27ba4d363eb56d2835f9903a

**Comment by jrieken** [comment]
There is `vscode.CompletionList.isIncomplete` for this
